src/app/organizations/state/collections.service.ts

```diff
@@ -162,51 +165,49 @@ export class CollectionsService {
-          // 2) on each collection summary on the org page
-          // So, this delete might have been invoked from the org page, and we're there already. Or not.
-          // Gracefully handle both cases, so that no matter how the UI evolves, this function works properly:
-          if (this.router.url.endsWith('/organizations/' + organizationName)) {
-            // We're already on the organization page.
-            // A router.navigate to the current page won't trigger the org component ngOnInit to update the state.
-            // Update the state manually.
-            this.updateCollections(organizationId);
-            // Organization has a collectionsLength property so we update it, too.
-            this.organizationService.updateOrganizationFromID(organizationId);
-          } else {
-            // Navigate to the organization page.
-            // Router.navigate will trigger the org component ngOnInit, which updates the necessary state.
-            this.router.navigate(['/organizations', organizationName]);
-          }
-        },
-        (error: HttpErrorResponse) => {
-          this.collectionsStore.setLoading(false);
-          this.collectionsStore.setError(true);
-          this.alertService.detailedError(error);
+    this.openApiOrganizationsService.deleteCollection(organizationId, collectionId).subscribe(
+      () => {
+        this.collectionsStore.setLoading(false);
+        this.collectionsStore.setError(false);
+        this.alertService.detailedSuccess();
+        this.matDialog.closeAll();
+        // Update and display the org page.
+        // There are at least two reasonable places for a delete collection button to appear in our UI:
+        // 1) on the collection page
+        // 2) on each collection summary on the org page
+        // So, this delete might have been invoked from the org page, and we're there already. Or not.
+        // Gracefully handle both cases, so that no matter how the UI evolves, this function works properly:
+        if (this.router.url.endsWith('/organizations/' + organizationName)) {
+          // We're already on the organization page.
+          // A router.navigate to the current page won't trigger the org component ngOnInit to update the state.
+          // Update the state manually.
+          this.updateCollections(organizationId);
+          // Organization has a collectionsLength property so we update it, too.
+          this.organizationService.updateOrganizationFromID(organizationId);
+        } else {
+          // Navigate to the organization page.
+          // Router.navigate will trigger the org component ngOnInit, which updates the necessary state.
+          this.router.navigate(['/organizations', organizationName]);
         }
-      );
+      },
+      (error: HttpErrorResponse) => {
+        this.collectionsStore.setLoading(false);
+        this.collectionsStore.setError(true);
+        this.alertService.detailedError(error);
+      }
+    );
```

Comment:
As far as I can tell, the above changes are probably due to a code reformat by our automatic tool.  Did anything functional change in `deleteCollection`?

Reply:
Yeah everything below my change on line 108 was automatic reformatting (makes the changes look so confusing on git).  No changes to `deleteCollection`